src/hamiltonian.jl

```diff
@@ -108,6 +108,10 @@ Base.isfinite(v::DualValue) = all(isfinite, v.value) && all(isfinite, v.gradient
 Base.isfinite(v::AbstractVecOrMat) = all(isfinite, v)
 Base.isfinite(z::PhasePoint) = isfinite(z.ℓπ) && isfinite(z.ℓκ)
 
+Base.isnan(v::DualValue) = any(isnan, v.value) || any(isnan, v.gradient)
+Base.isnan(v::AbstractVecOrMat) = any(isnan, v)
+Base.isnan(z::PhasePoint) = isnan(z.ℓπ) || isnan(z.ℓκ)
```

Comment:
Seems unrelated?

Reply:
Indeed, that's a hangover from #381 because I was a bit lazy to fix the commit history

Reply:
Realised this one has a double bump in Project.toml too. - let's decide on that PR first then I'll modify accordingly